trie/utils/nodes.py

```diff
@@ -89,3 +101,62 @@ def consume_common_prefix(left_key, right_key):
 
 def key_starts_with(full_key, partial_key):
     return all(left == right for left, right in zip(full_key, partial_key))
+
+
+# Binary Trie node utils
+def parse_node(node):
+    """
+    Input: a serialized node
+    """
+    if node is None or node == b'':
+        raise InvalidNode("Blank node is not a valid node type in Binary Trie")
+    elif node[0] == BRANCH_TYPE:
+        if len(node) != 65:
+            raise InvalidNode("Invalid branch node, both child node should be 32 bytes long each")
+        # Output: left child, right child, node type
+        return node[1:33], node[33:], BRANCH_TYPE
```

Comment:
I'm curious about the choice to return the type as the 3rd parameter.  Maybe my intuition is off but I would have expected it to be the first parameter.  Not sure why, but it feels more correct.  

Thoughts? (I'm not saying this is wrong, just curious if there is a reason for the current ordering)